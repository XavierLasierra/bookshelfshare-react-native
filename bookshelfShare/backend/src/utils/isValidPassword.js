const bcrypt = require("bcrypt");

function isValidPassword(password) {
  return bcrypt.compareSync(password, this.password);
}

module.exports = { isValidPassword };
