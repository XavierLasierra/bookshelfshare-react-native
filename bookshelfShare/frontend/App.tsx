import React, { useEffect } from 'react';
import { Provider } from 'react-redux';
import { NavigationContainer } from '@react-navigation/native';
import { MenuProvider } from 'react-native-popup-menu';
import SplashScreen from 'react-native-splash-screen';

import MainNavigator from './src/navigators/MainNavigator/MainNavigator';
import Notification from './src/components/Notification/Notification';

import configureStore from './src/redux/store';

export default function App() {
  useEffect(() => {
    SplashScreen.hide();
  }, []);

  return (
    <Provider store={configureStore()}>
      <MenuProvider>
        <NavigationContainer>
          <MainNavigator />
          <Notification />
        </NavigationContainer>
      </MenuProvider>
    </Provider>
  );
}
