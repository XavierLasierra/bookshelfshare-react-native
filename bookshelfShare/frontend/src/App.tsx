import React, {useEffect} from 'react';
import {Provider} from 'react-redux';
import {NavigationContainer} from '@react-navigation/native';
import {MenuProvider} from 'react-native-popup-menu';
import SplashScreen from 'react-native-splash-screen';

import MainNavigator from './navigators/MainNavigator/MainNavigator';
import Notification from './components/Notification/Notification';

import configureStore from './redux/store';

export default function App() {
  useEffect(() => {
    SplashScreen.hide();
  }, []);

  return (
    <Provider store={configureStore()}>
      <MenuProvider>
        <NavigationContainer>
          <MainNavigator />
          <Notification />
        </NavigationContainer>
      </MenuProvider>
    </Provider>
  );
}
