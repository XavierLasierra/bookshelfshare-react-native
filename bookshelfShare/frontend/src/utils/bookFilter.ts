export function bookFilter(text: string, books: any) {
  return books.reduce((acc: any, book: any) => {
    if (book.title.toLowerCase().includes(text.toLowerCase())) return [...acc, book];
    if (book.publisher.toLowerCase().includes(text.toLowerCase())) return [...acc, book];
    if (book.isbn.ISBN13.toLowerCase().includes(text.toLowerCase())) return [...acc, book];
    if (book.isbn.ISBN10.toLowerCase().includes(text.toLowerCase())) return [...acc, book];
    if (book.authors.some((author: string) => author.toLowerCase()
      .includes(text.toLowerCase()))) return [...acc, book];

    return acc;
  }, []);
}

export function bookShelfListFilter(text: string, books: any) {
  return books.reduce((acc: any, book: any) => {
    if (book.bookData.title.toLowerCase().includes(text.toLowerCase())) return [...acc, book];
    if (book.bookData.publisher.toLowerCase().includes(text.toLowerCase())) return [...acc, book];
    if (book.bookData.isbn.ISBN13.toLowerCase().includes(text.toLowerCase())) return [...acc, book];
    if (book.bookData.isbn.ISBN10.toLowerCase().includes(text.toLowerCase())) return [...acc, book];
    if (book.bookData.authors.some((author: string) => author.toLowerCase()
      .includes(text.toLowerCase()))) return [...acc, book];

    return acc;
  }, []);
}

export function bookShelfLocationFilter(filterLocation: number[], books: any) {
  return books.filter(({ customInformation: { location } }: any) => JSON.stringify(location)
  === JSON.stringify(filterLocation));
}
