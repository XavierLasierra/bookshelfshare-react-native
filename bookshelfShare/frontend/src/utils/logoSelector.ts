import BookIcon from '../assets/bookIcon.svg';
import SearchIcon from '../assets/searchIcon.svg';
import ProfileIcon from '../assets/profileIcon.svg';

export default function logoSelector(logoName: string) {
  switch (logoName) {
    case 'SearchIcon':
      return SearchIcon;
    case 'BookIcon':
      return BookIcon;
    case 'ProfileIcon':
      return ProfileIcon;
    default:
      return BookIcon;
  }
}
