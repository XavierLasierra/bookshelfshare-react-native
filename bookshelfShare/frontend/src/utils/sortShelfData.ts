export default function sortShelfData(shelf: any) {
  return shelf?.books.reduce((acc: any, book: any) => {
    const foundShelfBox = acc.find(({ location }: any) => JSON.stringify(location)
         === JSON.stringify(book?.customInformation?.location));
    if (foundShelfBox) {
      foundShelfBox.books.push(book.bookData);
    } else {
      acc.push({
        location: book?.customInformation?.location,
        books: [book.bookData]
      });
    }
    return acc;
  }, []);
}
