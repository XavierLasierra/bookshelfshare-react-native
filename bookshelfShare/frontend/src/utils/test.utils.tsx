import React from 'react';
import { render as rtlRender } from '@testing-library/react-native';
import { Provider } from 'react-redux';

import configureStore from '../redux/store';

function render(
  component: any,
  initialState: any
) {
  const store = configureStore(initialState);

  function Wrapper({ children }: any) {
    return (
      <Provider store={store}>
        {children}
      </Provider>
    );
  }
  return rtlRender(component, { wrapper: Wrapper });
}

// re-export everything
export * from '@testing-library/react-native';
// override render method
export { render };
