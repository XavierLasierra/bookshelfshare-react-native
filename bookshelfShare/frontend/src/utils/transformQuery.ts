interface Query {
    isbn?: string,
    inauthor?: string,
    intitle?: string,
    inpublisher?: string
}

export default function transformQuery(query: Query):string {
  const queryEntries = Object.entries(query);
  const url = queryEntries
    .reduce((acc, queryElement, index) => `${acc}${queryElement[0]}=${queryElement[1]}${index < queryEntries.length - 1 ? '+' : ''}`, '');
  return `${url}&key=${process.env.GOOGLE_API_KEY}`;
}
