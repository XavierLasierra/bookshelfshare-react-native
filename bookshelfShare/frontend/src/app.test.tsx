import React from 'react';
import SplashScreen from 'react-native-splash-screen';
import App from './App';
import { render } from './utils/test.utils';

jest.mock('react-native-splash-screen', () => ({
  hide: jest.fn()
}));

jest.useFakeTimers();

describe('Given an App component', () => {
  describe('When it is triggered', () => {
    let screen: any;
    beforeEach(() => {
      screen = render(<App />);
    });
    test('Then should match the snapshot', () => {
      expect(screen).toMatchSnapshot();
    });

    test('Then SplashScreen.hide should have been called', () => {
      expect(SplashScreen.hide).toHaveBeenCalled();
    });
  });
});
