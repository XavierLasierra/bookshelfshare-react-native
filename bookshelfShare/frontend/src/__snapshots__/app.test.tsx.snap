// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Given an App component When it is triggered Then should match the snapshot 1`] = `
<View
  onLayout={[Function]}
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    style={
      Array [
        Object {
          "flex": 1,
        },
        undefined,
        undefined,
        undefined,
      ]
    }
  >
    <View
      style={
        Object {
          "flex": 1,
        }
      }
    >
      <RNCSafeAreaProvider
        onInsetsChange={[Function]}
        style={
          Array [
            Object {
              "flex": 1,
            },
            undefined,
          ]
        }
      >
        <View
          style={
            Array [
              Object {
                "backgroundColor": "rgb(242, 242, 242)",
                "flex": 1,
              },
              undefined,
            ]
          }
        >
          <View
            onLayout={[Function]}
            style={
              Object {
                "flex": 1,
              }
            }
          >
            <View
              pointerEvents="box-none"
              style={
                Object {
                  "bottom": 0,
                  "left": 0,
                  "position": "absolute",
                  "right": 0,
                  "top": 0,
                }
              }
            >
              <View
                collapsable={false}
                style={
                  Object {
                    "opacity": 1,
                  }
                }
              />
              <View
                accessibilityElementsHidden={false}
                closing={false}
                gestureVelocityImpact={0.3}
                importantForAccessibility="auto"
                onClose={[Function]}
                onGestureBegin={[Function]}
                onGestureCanceled={[Function]}
                onGestureEnd={[Function]}
                onOpen={[Function]}
                onTransition={[Function]}
                pointerEvents="box-none"
                style={
                  Array [
                    Object {
                      "display": "flex",
                      "overflow": undefined,
                    },
                    Object {
                      "bottom": 0,
                      "left": 0,
                      "position": "absolute",
                      "right": 0,
                      "top": 0,
                    },
                  ]
                }
                transitionSpec={
                  Object {
                    "close": Object {
                      "animation": "spring",
                      "config": Object {
                        "damping": 500,
                        "mass": 3,
                        "overshootClamping": true,
                        "restDisplacementThreshold": 10,
                        "restSpeedThreshold": 10,
                        "stiffness": 1000,
                      },
                    },
                    "open": Object {
                      "animation": "spring",
                      "config": Object {
                        "damping": 500,
                        "mass": 3,
                        "overshootClamping": true,
                        "restDisplacementThreshold": 10,
                        "restSpeedThreshold": 10,
                        "stiffness": 1000,
                      },
                    },
                  }
                }
              >
                <View
                  collapsable={false}
                  nativeID="animatedComponent"
                  pointerEvents="box-none"
                  style={
                    Object {
                      "flex": 1,
                      "marginTop": 0,
                    }
                  }
                >
                  <View
                    enableTrackpadTwoFingerGesture={true}
                    enabled={false}
                    hitSlop={
                      Object {
                        "right": -700,
                      }
                    }
                    maxDeltaY={20}
                    minOffsetX={5}
                    onHandlerStateChange={[Function]}
                  >
                    <View
                      needsOffscreenAlphaCompositing={true}
                      style={
                        Object {
                          "flex": 1,
                          "opacity": 1,
                          "transform": Array [
                            Object {
                              "translateY": 0,
                            },
                          ],
                        }
                      }
                    >
                      <View
                        style={
                          Array [
                            Object {
                              "flex": 1,
                              "overflow": "hidden",
                            },
                            Array [
                              Object {
                                "backgroundColor": "rgb(242, 242, 242)",
                              },
                              undefined,
                            ],
                          ]
                        }
                      >
                        <View
                          style={
                            Object {
                              "flex": 1,
                              "flexDirection": "column-reverse",
                            }
                          }
                        >
                          <View
                            style={
                              Object {
                                "flex": 1,
                              }
                            }
                          >
                            <View
                              collapsable={false}
                              style={
                                Object {
                                  "flex": 1,
                                }
                              }
                            >
                              <RCTSafeAreaView
                                emulateUnlessSupported={true}
                                style={
                                  Object {
                                    "alignItems": "center",
                                    "backgroundColor": "#D9D8C4",
                                    "display": "flex",
                                    "height": "100%",
                                    "justifyContent": "center",
                                  }
                                }
                              >
                                <View
                                  style={
                                    Object {
                                      "alignItems": "center",
                                      "height": 30,
                                    }
                                  }
                                >
                                  <View
                                    collapsable={false}
                                  >
                                    <IconMock
                                      height={60}
                                      width={60}
                                    />
                                  </View>
                                </View>
                              </RCTSafeAreaView>
                            </View>
                          </View>
                        </View>
                      </View>
                    </View>
                  </View>
                </View>
              </View>
            </View>
          </View>
          <View
            pointerEvents="box-none"
            style={
              Object {
                "height": 44,
                "left": 0,
                "position": "absolute",
                "right": 0,
                "top": 0,
                "zIndex": 1,
              }
            }
          />
        </View>
      </RNCSafeAreaProvider>
    </View>
    <View
      pointerEvents="none"
      style={
        Object {
          "bottom": 0,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        }
      }
    />
  </View>
  <RCTSafeAreaView
    emulateUnlessSupported={true}
    pointerEvents="box-none"
    style={
      Object {
        "bottom": 0,
        "left": 0,
        "position": "absolute",
        "right": 0,
        "top": 0,
      }
    }
  >
    <View
      collapsable={false}
      onLayout={[Function]}
      pointerEvents="box-none"
      style={
        Object {
          "flex": 1,
        }
      }
    />
  </RCTSafeAreaView>
</View>
`;
